src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -445,22 +451,21 @@ trait EntityComponent {
     }
 
     def cloneEntities(destWorkspaceContext: SlickWorkspaceContext, entities: TraversableOnce[Entity]): ReadWriteAction[Unit] = {
-      val entityInserts = batchInsertEntities(destWorkspaceContext, entities.toSeq.map(e => marshalNewEntity(e, destWorkspaceContext.workspaceId)))
-
-      val attributeInserts = entityInserts flatMap { ids =>
+      batchInsertEntities(destWorkspaceContext, entities.toSeq.map(e => marshalNewEntity(e, destWorkspaceContext.workspaceId))) flatMap { ids =>
         val entityIdByEntity = ids.map(record => record.id -> entities.filter(p => p.entityType == record.entityType && p.name == record.name).toSeq.head)
-        val entityIdsByRef = entityIdByEntity.map{ case (entityId, entity) => entity.toReference -> entityId}.toMap
-
-        val attributeRecords = entityIdByEntity flatMap { case (entityId, entity) =>
-          entity.attributes.toIterable.flatMap { case (name, attr) =>
-            entityAttributeQuery.marshalAttribute(entityId, name, attr, entityIdsByRef)
+        val entityIdsByRef = entityIdByEntity.map { case (entityId, entity) => entity.toReference -> entityId }.toMap
+
+        attributeNamespaceQuery.getMap flatMap { attributeNamespaceMapping =>
```

Comment:
getMap instead of getId here, to get them all at once.  Can't use this further down because those flatMaps are on regular collections, not DBIO Actions!